### ISSUE TITLE
tutorial on how autoscaling works

### ISSUE THREAD

**Issue by tacaswell** [matplotlib/matplotlib#17417]
> But I'd settle for some better/less tautological documentation for how autoscale is supposed to work :)

_Originally posted by @mwaskom in https://github.com/matplotlib/matplotlib/issues/17331#issuecomment-624125670_

We need a long-form tutorial like https://matplotlib.org/tutorials/intermediate/imshow_extent.html for autoscaling.  This needs to cover

 - margins
 - 'sticky edges'
 - what artists participate
 - when / how the bounding boxes are collected
 - what implicitly disables autoscaling
 - how to explicitly enable / disable autoscaling
 - when and how you need to manually kick off the autoscaling 
 - how to "manually" get the same effect

**Comment by Tillsten** [comment]
I would add:
  * How to autoscale one axis only within the view limits of the other coordinate.

**Comment by sarthakforwet** [comment]
Hey, I wanna work on this issue.. Could you please assign it to me?
And also I am quite not getting what 

> what artists participate  

means.

**Comment by tacaswell** [comment]
@sarthakforwet If you are interested in working on this please do so and open a PR, we do not really "assign" work.

Not all of the artists associated with the axes are taken into account when deciding what the data limits are, see https://github.com/matplotlib/matplotlib/blob/fb9e72309aeefb51d1a56ecbdb1e7399d105ff06/lib/matplotlib/axes/_base.py#L2100-L2127

**Comment by sarthakforwet** [comment]
Hey @tacaswell . I know autoscaling is not supported by Collection objects such as EllipseCollection,LineCollection,PatchCollection etc. but  certainly not getting the answer for what implicitly disables it. 

>o how to explicitly enable / disable autoscaling
>o when and how you need to manually kick off the autoscaling

Are you trying to say that in the first one describe for `pyplot.autoscale(enable = True/False)`  and
in the second discuss a use-case when we want not to autoscale and hence disable it?

> o how to "manually" get the same effect

By manually getting the same effect you mean setting xlim and ylim manually to scale according to data right ?

**Comment by tacaswell** [comment]
>  how to explicitly enable / disable autoscaling

Yes, but the tutorial should start at the `Axes` level

> when and how you need to manually kick off the autoscaling

If you update the data in an artist you may need to "manually" update what the Axes thinks its data limits are

>  how to "manually" get the same effect

I was thinking what series of methods would the user need to call to replicate what autoscaling does when a new artist is added, but going through how to, given a set of artists, compute what the limits would be also sounds super useful!

**Comment by sarthakforwet** [comment]
@tacaswell  Okay and what implicitly disables autoscalling?

**Comment by sarthakforwet** [comment]
@tacaswell  I have made the tutorial but some things are not covered and I have searched but did not get how to do that so should i submit the PR for you to review it or first share it via other medium and finally submit a completed tutorial ?

**Comment by tacaswell** [comment]
We try to run all of our contributions / review through github so please open a (draft) PR.

Some of these things are not currently documented so part of the work here is to dig into the source and describe what it is doing ;)

**Comment by ianhi** [comment]
I have two thoughts to add

>          At present, `~.Collection` instances are not supported. 

1. Can what `relim` is/does also be added to such a tutorial. I don't really understand the difference or how I could achieve something similar with a collection. 

2. It would be cool if https://matplotlib.org/gallery/api/collections.html was updated in concert with this. I found that reading that has muddied the waters on relim/autoscale for me. Sometimes autoscale_view does apply to collections, but maybe not relim? And there is a difference between `offsets` and `transOffsets`?
